src/ui/public/vis/editors/default/default.js

```diff
@@ -64,19 +64,24 @@ const defaultEditor = function ($rootScope, $compile) {
 
           updateScope();
 
+          $scope.state = $scope.vis.copyCurrentState();
+          $scope.oldState = $scope.vis.getSerializableState($scope.state);
+
           $scope.toggleSidebar = () => {
             $scope.$broadcast('render');
           };
 
           this.el.one('renderComplete', resolve);
-
           // track state of editable vis vs. "actual" vis
           $scope.stageEditableVis = () => {
+            $scope.vis.setCurrentState($scope.state);
```

Comment:
If there is no specific reason for the order of these two lines, I would rather first get the `oldStat` and then start setting the new one (and updating it), since it feels a bit more natural to read imho. I looked at this and wondered at first why we set the current state before actually getting the old state.

Reply:
we first need to set the vis state and the reset the old state to vis state